app/frontend/src/features/auth/useAuthStore.ts

```diff
@@ -111,13 +116,12 @@ export default function useAuthStore() {
         setError(null);
         setLoading(true);
         try {
-          const isJailed = (await service.jail.getIsJailed()).isJailed;
-          setJailed(isJailed);
-          if (!isJailed) {
-            ///TODO: Remove this when changing to userId
-            const user = await service.user.getCurrentUser();
-            setUser(user);
+          const res = await service.jail.getIsJailed();
+          if (!res.isJailed) {
+            setUserId(res.user!.userId);
```

Comment:
We can avoid the non-null assertion here if we re-write the bit that returns `res` in `service.jail.getIsJailed` like this:
```typescript
// in jail.ts
export async function getIsJailed() {
  // Some other bits of code before that...
  if (!isJailed) {
    const user = await getCurrentUser();
    return { isJailed, user };
  }
  return { isJailed, user: null };
}
```
Since that changes the inferred return type from `Promise<{ isJailed: boolean; user: User.AsObject | null; }>` to `Promise<{ isJailed: true; user: null } | { isJailed: false; user: User.AsObject }>` which creates a [discriminating union type](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions), so when you do the if check here, TS correctly narrows the type again and knows that `res.user` exists if `isJailed` is `false`